precompiles/xcm/src/lib.rs

```diff
@@ -280,6 +291,21 @@ where
             return Err(revert("Assets resolution failure."));
         }
 
+        let value = handle.context().apparent_value;
+        if value > U256::zero() {
+            handle.check_function_modifier(FunctionModifier::NonPayable)?;
```

Comment:
Seems redundant if you've already ensured these checks are made in the `execute` function.

Reply:
Indeed.